### PR TITLE
Make it easier to view and compare code coverage for a file

### PR DESCRIPTION
**Pull request by JamesWTruher**
Fix the default path for finding the code coverage build to use `win7-x64` rather than `win10-x64`
Move the test results files to `$HOME/Documents` rather than `$PWD` which will remove the annoying `git status` showing these two files after getting code coverage. 
Change the default path for the coverage file for `Get-CodeCoverage` to be the location that `Invoke-OpenCover` uses.
Change the file coverage tool to take output from the GetFileCoverage Method. The earlier process was way to hard. Now you can type:
```powershell
$CodeCoverage = Get-CodeCoverage
$CodeCoverage.GetFileCoverage("Program.cs") | Format-FileCoverage
```
and see the lines covered and missed in any file named `Program.cs`
You can now compare coverage for a file between two runs with Compare-FileCoverage
```powershell
PS# compare-filecoverage $old $new program.cs

FileName   ReferenceCoverage DifferenceCoverage CoverageDelta
--------   ----------------- ------------------ -------------
Program.cs 100.00            100.00             0.00
```

### CODE REVIEW

File: test/tools/OpenCover/OpenCover.psm1 [hunk 1]
```diff
@@ -506,13 +506,47 @@ function Compare-CodeCoverage
...
+    # so we can compare file coverage
+    $reference = $ReferenceCoverage.GetFileCoverage($FileName) | Foreach-Object { $h = @{} } { $h[$_.path] = $_ } {$h}
+    $difference = $differenceCoverage.GetFileCoverage($FileName)| Foreach-Object { $h = @{}}{ $h[$_.path] = $_ }{$h }
+    # based on the paths, create objects which show the difference between the two runs
+    $reference.Keys | Sort-Object | Foreach-Object {
+        $referenceObject = $reference[$_]
+        $differenceObject = $difference[$_]
+        if ( $differenceObject )
+        {
+            $o = [pscustomobject]@{
```
Comment (TravisEz13) [inline test/tools/OpenCover/OpenCover.psm1]: nit: `$o` could have a more meaningful name

File: test/tools/OpenCover/OpenCover.psm1 [hunk 2]
```diff
@@ -506,13 +506,47 @@ function Compare-CodeCoverage
...
+{
+    param (
+        [Parameter(Position=0,Mandatory=$true)]$ReferenceCoverage,
+        [Parameter(Position=1,Mandatory=$true)]$DifferenceCoverage,
+        [Parameter(Position=2,Mandatory=$true)]$FileName
+    )
+    # create a couple of hashtables where the key is the path
+    # so we can compare file coverage
+    $reference = $ReferenceCoverage.GetFileCoverage($FileName) | Foreach-Object { $h = @{} } { $h[$_.path] = $_ } {$h}
+    $difference = $differenceCoverage.GetFileCoverage($FileName)| Foreach-Object { $h = @{}}{ $h[$_.path] = $_ }{$h }
```
Comment (SteveL-MSFT) [inline test/tools/OpenCover/OpenCover.psm1]: should have a space before the pipe

**Review by SteveL-MSFT — APPROVED** [review]
Minor issue, but not blocking

File: test/tools/OpenCover/OpenCover.psm1 (hunk 1 above)
Comment (JamesWTruher) [reply test/tools/OpenCover/OpenCover.psm1]: quite right - updated to be more descriptive

---
In reply to: [137401232](https://github.com/PowerShell/PowerShell/pull/4764#discussion_r137401232) [](ancestors = 137401232)

File: test/tools/OpenCover/OpenCover.psm1 (hunk 2 above)
Comment (JamesWTruher) [reply test/tools/OpenCover/OpenCover.psm1]: fixed

---
In reply to: [137402201](https://github.com/PowerShell/PowerShell/pull/4764#discussion_r137402201) [](ancestors = 137402201)

**Review by TravisEz13 — APPROVED** [review]

**Comment by JamesWTruher** [comment]
I'm looking for a quick turn around on this because I would like to use this update Thursday and Friday rather than requiring everyone use my fork.

**Pull request merged**